test/dlc_tests/ec2/test_gdrcopy.py

```diff
@@ -0,0 +1,40 @@
+import os
+import pytest
+
+import test.test_utils as test_utils
+from test.test_utils import CONTAINER_TESTS_PREFIX, is_pr_context, is_efa_dedicated
+from test.test_utils.ec2 import (
+    get_efa_ec2_instance_type,
+    filter_efa_instance_type,
+    execute_ec2_training_test,
+)
+
+GDRCOPY_SANITY_TEST_CMD = os.path.join(CONTAINER_TESTS_PREFIX, "gdrcopy", "test_gdrcopy.sh")
+EC2_EFA_GPU_INSTANCE_TYPE_AND_REGION = get_efa_ec2_instance_type(
+    default="p4d.24xlarge",
+    filter_function=filter_efa_instance_type,
+)
+
+
+@pytest.mark.usefixtures("sagemaker")
+@pytest.mark.processor("gpu")
+@pytest.mark.model("N/A")
+@pytest.mark.integration("gdrcopy")
+@pytest.mark.parametrize("ec2_instance_type,region", EC2_EFA_GPU_INSTANCE_TYPE_AND_REGION)
+@pytest.mark.parametrize(
+    "ec2_instance_ami", [test_utils.UBUNTU_20_BASE_DLAMI_US_WEST_2], indirect=True
+)
+@pytest.mark.skipif(
+    is_pr_context() and not is_efa_dedicated(),
+    reason="Skip GDRCopy test in PR context unless explicitly enabled",
+)
+def test_gdrcopy(pytorch_training, ec2_connection, ec2_instance_type, region, gpu_only):
+    if test_utils.is_image_incompatible_with_instance_type(pytorch_training, ec2_instance_type):
+        pytest.skip(
+            f"Image {pytorch_training} is incompatible with instance type {ec2_instance_type}"
+        )
+    if test_utils.is_below_framework_version("1.13.1", pytorch_training, "pytorch"):
+        pytest.skip(f"GDRCopy is not supported on Image {pytorch_training}")
```

Comment:
The use `ec2_connection` as a pytest parameter requires that the instance already be created and set up for the test to start running. Therefore, when this test needs to be skipped, it creates an instance and then skips the test, instead of skipping the setup-phase completely.

The only way to skip the setup-phase is to prevent the `pytorch_training` parameter from being assigned the PT<=1.12 image URI in the first place.

Please refer to https://github.com/aws/deep-learning-containers/blob/0f10e9b1dabebcafc0a786d3b5915fcdfa9a03dc/test/dlc_tests/conftest.py#L986-L988
https://github.com/aws/deep-learning-containers/blob/0f10e9b1dabebcafc0a786d3b5915fcdfa9a03dc/test/dlc_tests/conftest.py#L1086-L1089
and
https://github.com/aws/deep-learning-containers/blob/0f10e9b1dabebcafc0a786d3b5915fcdfa9a03dc/test/dlc_tests/ec2/pytorch/inference/test_pytorch_inference.py#L170-L172

to please add a `pt113_and_above_only` fixture that lets this test be skipped on versions that shouldn't be tested.